src/gateway.c

```diff
@@ -513,10 +519,11 @@ static int handle_exec(struct gateway *g, struct handle *req)
 	}
 	req->stmt_id = stmt->id;
 	g->req = req;
-	req_id = idNext(&g->random_state);
-	rv = leader__exec(g->leader, &g->exec, stmt->stmt, req_id,
-			  leader_exec_cb);
-	if (rv != 0) {
+	rv = leader_exec_v2(g->leader, &g->exec, stmt->stmt, leader_exec_cb);
+	if (rv == LEADER_NOT_ASYNC) {
+		/* XXX */
```

Comment:
?

Reply:
Left over from experimenting with different ways to write the new "invoke callback manually" codepaths, no defect that I forgot to remove afaict.